apso/src/main/scala/eu/shiftforward/apso/io/FileDescriptor.scala

```diff
@@ -52,6 +52,16 @@ trait FileDescriptor {
   def upload(localTarget: LocalFileDescriptor): Boolean
 
   /**
+   * Uploads an input stream to this file's location.
+   * The target must point to a file.
+   *
+   * @param inputStream the input stream that should be uploaded.
+   * @param length the input stream length (leaving this as `None` can have performance implications)
+   * @return `true` if the upload was successful, `false` otherwise.
+   */
+  def upload(inputStream: InputStream, length: Option[Long]): Boolean
```

Comment:
I would rather leave this as is for now, as not defining the length when uploading to S3 can lead to OOM errors.

We can eventually change this if we add new file descriptors that do not require the length.